py-polars/polars/_utils/construction/dataframe.py

```diff
@@ -655,9 +655,12 @@ def _sequence_of_dict_to_pydf(
         if column_names
         else None
     )
-    pydf = PyDataFrame.from_dicts(
-        data, infer_schema_length, dicts_schema, schema_overrides
-    )
+    if isinstance(data, pl.Series) and data.dtype == Struct:
+        pydf = data.struct.unnest()._df
```

Comment:
This isn't quite correct because we aren't taking into account the schema.

Reply:
Will add a test to check that, and adjust as needed; the subsequent `_post_apply_columns` will handle any `schema_overrides`, but need to check how `schema` interacts. The `unnest` approach will be a _massive_ speedup, so we should preserve it.

Reply:
> This isn't quite correct because we aren't taking into account the schema. 

Confirmed; it actually does do the right thing here; the "schema" value is consolidated with "schema_overrides" by `_unpack_schema`, so `_post_apply_columns` handles any required casts. Adding a unit test to validate.

> We should defer to `series_to_pydf`.

This would create a very different result; instead of unpacking the records as columns, you'd get a frame containing a single Struct column ;)